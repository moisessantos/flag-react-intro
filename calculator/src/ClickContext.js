import React from 'react';

export const ClickContext = React.createContext();

export const useClickContext = () => {
  const context = React.useContext(ClickContext);
  if (!context) {
    throw new Error('useClickContext must be used within the ClickContextProvider');
  }
  return context;
};
