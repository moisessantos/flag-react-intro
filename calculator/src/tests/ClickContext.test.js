import React from 'react';

import { useClickContext, ClickContext } from '../ClickContext';

jest.mock('react');

describe('<useClickContext /> spec', () => {
  it('returns context as expected', () => {
    React.useContext = jest.fn(() => 'ClickContext');
    expect(useClickContext()).toBe('ClickContext');
    expect(React.useContext).toBeCalledWith(ClickContext);
  });

  it('throws error when context is undefined', () => {
    React.useContext = jest.fn(() => null);
    try {
      useClickContext();
    } catch (ex) {
      expect(ex).toMatchSnapshot();
    }
  });
});
