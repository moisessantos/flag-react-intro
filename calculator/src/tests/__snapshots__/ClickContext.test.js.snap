// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<useClickContext /> spec throws error when context is undefined 1`] = `[Error: useClickContext must be used within the ClickContextProvider]`;
